src/Security/Group.php

```diff
@@ -522,16 +513,6 @@ public function validate()
             ->map('Code', 'Title')
             ->toArray();
 
-        if (isset($currentGroups[$this->Code])) {
```

Comment:
I'm just wondering if it's the correct approach to delete validation and just increment groups?  

Is the reason why we are deleting the validation is because it's just never worked?  Or it used to work and we accidentally broke it with a recent PR?

If it never worked then it makes sense to delete this code block.  Otherwise wouldn't the correct fix here be to fix the broken validation rather than incrementing duplicate groups?

(I can see how enforcing validation when programatically setting a group would be difficult)

Reply:
> delete validation and just increment groups

That's what's already happening. This validation code doesn't get reached at all because the code is already incremented in `setCode()`.

> Is the reason why we are deleting the validation is because it's just never worked? Or it used to work and we accidentally broke it with a recent PR?

Yes, we're deleting it because it never worked. It was added at the same time as the code in `setCode()` which removed deduping. Now that I'm moving the deduping logic to `onBeforeWrite()` the validation _would_ kick in if I kept it here, and the deduping wouldn't happen, which would be a breaking change.
Based on the comments in the issue and in the original PR, it sounds like the idea was always to add deduping - there's no clear reason why this validation code was added in the first place but it just wasn't being called in any case.